### ISSUE TITLE
"Decorator" modules resolvers ignored on properties they do not define

### ISSUE THREAD

**Issue by russell-dot-js** [Urigo/graphql-modules#725]
Hi all,

First off, I'm loving graphql-modules! Been in production with it for almost a year.

However, I recently updated a project (quite a few version numbers) from 0.4.2 to 0.7.13

This update caused some of our code (using what I'm calling "Decorator modules") to break.

I will try to simplify how to reproduce, please let me know if you have a better way of reporting issues.

**Module A**
```
export const FooModule new GraphQLModule({
    typeDefs: gql`
      type Foo {
        foo: String!
      }
    `
});
```

**Module B**
```
export const BarModule new GraphQLModule({
    imports: [FooModule],
    resolvers: {
      Foo: {
        foo: () => 'bar',
      }
    }
});
```

In 0.4.2, Foo.foo would always be "bar". Now, it is null, and it will even fail if BarModule also defines Foo.foo as a String. 

E.g. (this hack does not fix the issue):

```
export const BarModule new GraphQLModule({
    imports: [FooModule],
    typeDefs: gql`
      type Foo {
        foo: String!
      }
    `,
    resolvers: {
      Foo: {
        foo: () => 'bar',
      }
    }
});
```

We use this in a few places, e.g. if we extend a type exposed in Module A in Module B, and Module B needs to access a parent property to resolve, we will do all of the decoration in Module B (e.g. passing an accountId from the top-level object down to a deeply nested child object) as this logic is specific to Module B. To us it makes sense to keep this logic in Module B, as it kind of breaks the spirit of encapsulation if Module A needs to know to decorate all of its types nested properties with data needed for other Modules to function correctly.

So it seems to me like graphql-modules now ignores resolvers for types that are not initially defined in the same module, whereas they used to be invoked.

Can you please let me know if this change was intentional, and we need to update our code to no longer use this pattern, having Module A decorate all of its types' child properties so that dependent modules can access them, or was this a regression? ~~If I've stumbled upon a regression, can you please let me know which version I can downgrade to in order to bring back the desired functionality?~~ Downgrading to 0.65 resolves this issue

Thank you!

**Comment by russell-dot-js** [comment]
I've confirmed that version 0.6.5 works, 0.7.0 does not work, 0.6.6 untestable as it has some other error related to values added to context

**Comment by russell-dot-js** [comment]
I haven't pinpointed the exact cause, but am taking an educated guess that the behavior changed with: https://github.com/Urigo/graphql-modules/pull/375/files

**Comment by ardatan** [comment]
@russell-dot-js You're right, the behavior should be exactly how you wrote. I will add a reproduction test first then try to fix it.

**Comment by ardatan** [comment]
I tried to reproduce it but it succeeds
https://github.com/Urigo/graphql-modules/pull/726/files
Let me know if I miss something.

**Comment by russell-dot-js** [comment]
> I tried to reproduce it but it succeeds
> https://github.com/Urigo/graphql-modules/pull/726/files
> Let me know if I miss something.

Awesome, thank you so much for checking this out so quickly. I will fork and get you a repo

**Comment by russell-dot-js** [comment]
I was able to repro by adding FooModule as a dependency to the root module:
https://github.com/Urigo/graphql-modules/pull/728

**Comment by russell-dot-js** [comment]
Hi @ardatan, I just wanted to check in and see if there were any updates on this issue?

**Comment by kamilkisiela** [comment]
We released a new major version (v1.0) of GraphQL Modules. Here's the [migration guide](https://graphql-modules.com/docs/next/recipes/migration).

The new version solves the issue, so I'm closing the thread.

Please try to upgrade. If you can't and still want to use the legacy version, please open a new issue.
